File: BinarySearch/src/main/java/Solution162.java
public class Solution162 {
    public int findPeakElement(int[] nums) {
        int left = 0, right = nums.length - 1;
        while (left < right) {
            int lmid = (left + right) / 2;
            int rmid = lmid + 1;

            if (nums[rmid] <= nums[lmid]) {
                right = lmid;
            } else {
                left = lmid + 1;
            }
        }
        return right;

    }

    public static void main(String[] args) {
        Solution162 s = new Solution162();

        // int[] nums = { 1, 2, 3, 1 }; // 2
        int[] nums = { 1, 2, 1, 3, 5, 6, 4 }; // 1 or 5
        int res = s.findPeakElement(nums);
        System.out.println(res);
    }
}

File: BinarySearch/src/main/python/solution162.py
from typing import List


class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        left, right = 0, len(nums) - 1

        while left < right:
            left_mid = (left + right) // 2
            right_mid = left_mid + 1

            if nums[left_mid] >= nums[right_mid]:
                right = left_mid

            else:
                left = left_mid + 1

        return right